msg/templates/uorb_microcdr/microRTPS_client.cpp.em

```diff
@@ -86,54 +88,41 @@ void* send(void* /*unused*/)
     uint16_t header_length = 0;
 
     /* subscribe to topics */
-    int fds[@(len(send_topics))] = {};
-
-    // orb_set_interval statblish an update interval period in milliseconds.
 @[for idx, topic in enumerate(send_topics)]@
-    fds[@(idx)] = orb_subscribe(ORB_ID(@(topic)));
-    orb_set_interval(fds[@(idx)], _options.update_time_ms);
+    uORB::Subscription @(topic)_sub{ORB_ID(@(topic))};
 @[end for]@
 
     // ucdrBuffer to serialize using the user defined buffer
     ucdrBuffer writer;
-    header_length=transport_node->get_header_length();
-    ucdr_init_buffer(&writer, (uint8_t*)&data_buffer[header_length], BUFFER_SIZE - header_length);
+    header_length = transport_node->get_header_length();
+    ucdr_init_buffer(&writer, reinterpret_cast<uint8_t*>(&data_buffer[header_length]), BUFFER_SIZE - header_length);
 
     struct timespec begin;
     px4_clock_gettime(CLOCK_REALTIME, &begin);
 
     while (!_should_exit_task)
     {
-        bool updated;
 @[for idx, topic in enumerate(send_topics)]@
-        orb_check(fds[@(idx)], &updated);
-        if (updated)
-        {
-            // obtained data for the file descriptor
-            struct @(send_base_types[idx])_s data;
-            // copy raw data into local buffer
-            if (orb_copy(ORB_ID(@(topic)), fds[@(idx)], &data) == 0) {
-                /* payload is shifted by header length to make room for header*/
-                serialize_@(send_base_types[idx])(&writer, &data, &data_buffer[header_length], &length);
-
-                if (0 < (read = transport_node->write((char)@(rtps_message_id(ids, topic)), data_buffer, length)))
+        @(send_base_types[idx])_s @(topic)_data;
+        if (@(topic)_sub.update(&@(topic)_data)) {
+                // copy raw data into local buffer. Payload is shifted by header length to make room for header
+                serialize_@(send_base_types[idx])(&writer, &@(topic)_data, &data_buffer[header_length], &length);
+                if (0 < (read = transport_node->write(static_cast<char>(@(rtps_message_id(ids, topic))), data_buffer, length)))
                 {
                     total_sent += read;
                     ++sent;
                 }
-            }
         }
 @[end for]@
-
-        px4_usleep(_options.sleep_ms*1000);
+        px4_usleep(_options.sleep_ms * 1e3);
         ++loop;
     }
 
     struct timespec end;
     px4_clock_gettime(CLOCK_REALTIME, &end);
-    double elapsed_secs = double(end.tv_sec - begin.tv_sec) + double(end.tv_nsec - begin.tv_nsec)/double(1000000000);
+    double elapsed_secs = static_cast<double>(end.tv_sec - begin.tv_sec + (end.tv_nsec - begin.tv_nsec) / 1e9);
```

Comment:
`static_cast<double>` is ineffective here as the division by `1e9` already implicitly converts to double.

Reply:
Agreed